albumentations/core/transforms_interface.py

```diff
@@ -38,6 +57,158 @@
+        raise NotImplementedError
+
+    @abstractmethod
+    def __setitem__(self, key, value):
+        raise NotImplementedError
+
+    @abstractmethod
+    def check_consistency(self):
+        raise NotImplementedError
+
+    @staticmethod
+    @abstractmethod
+    def assert_array_format(array):
+        raise NotImplementedError
+
+
+@dataclass
+class BBoxesInternalType(BatchInternalType):
+    array: BoxesArray = field(default=np.empty((0, 4)))
+
+    def __eq__(self, other):
+        if not isinstance(other, BBoxesInternalType):
+            raise TypeError(
+                "`BBoxesInternalType` is only comparable with another `BBoxesInternalType`, "
+                f"given {type(other)} instead."
+            )
+        if not len(self.array) and not len(other.array) and len(self.targets) == len(other.targets) == 0:
+            # This's because numpy does not treat array([], dtype=float64)
+            # and array=array([], shape=(0, 4), dtype=float64) equally.
+            return True
+        return np.array_equal(self.array, other.array) and self.targets == other.targets
+
+    def __getitem__(self, item) -> "BBoxesInternalType":
+        _bboxes = self.array[item].astype(float)
+        if isinstance(item, int):
+            _bboxes = _bboxes[np.newaxis, ...]
+            _target = [self.targets[item]] if self.targets else []
+        elif isinstance(item, slice):
+            _target = self.targets[item]
+        else:
+            _target = [self.targets[i] for i in item] if self.targets else []
+        return BBoxesInternalType(array=_bboxes, targets=_target)
+
+    def __setitem__(self, idx, value: "BBoxesInternalType"):
+        self.array[idx] = value.array
+        if isinstance(idx, int):
+            self.targets[idx] = value.targets[0]
+        else:
+            for i, target in zip(idx, value.targets):
+                self.targets[i] = target
+
+    @staticmethod
+    def assert_array_format(bboxes: np.ndarray):  # noqa
+        if not isinstance(bboxes, np.ndarray):
+            raise TypeError("Bboxes should be a numpy ndarray.")
+        if len(bboxes):
+            if not (len(bboxes.shape) == 2 and bboxes.shape[-1] == 4):
+                raise ValueError(
+                    "An array of bboxes should be 2 dimension, and the last dimension must has 4 elements. "
+                    f"Received {bboxes.shape}."
+                )
+
+    def check_consistency(self):
+        if len(self.array) != len(self.targets):
+            raise ValueError(
+                "The amount of bboxes and additional targets should be the same. "
+                f"Get {len(self.array)} bboxes and {len(self.targets)} additional targets."
+            )
+        self.assert_array_format(self.array)
+
+
+@dataclass
+class KeypointsInternalType(BatchInternalType):
+    array: KeypointsArray = field(default=np.empty((0, 4)))
+
+    def __eq__(self, other):
+        if not isinstance(other, KeypointsInternalType):
```

Comment:
Maybe we could write in this manner?
```python
class Floats4InternalType(BatchInternalType):
    ...  # implementation

class KeypointsInternalType(Floats4InternalType):
    pass

class BBoxesInternalType(Floats4InternalType):
    pass
```

Reply:
I'm not sure I fully understand this.
you mean something like moving all actual implementation to `Floats4InternalType` and keep `KeypointsInternalType` and `BBoxesInternalType` with no overwritten methods?
then except `assert_array_format` and `check_consistency`, all the other methods would share the same implementation, something like this
```python
class Floats4InternalType(BatchInternalType):
    def __eq__(self, other):
        if not isinstance(other, self.__class__):
            raise TypeError(
                f"`{self.__class__}` is only comparable with another `{self.__class__}`, "
                f"given {type(other)} instead."
            )
        if not len(self.array) and not len(other.array) and len(self.targets) == len(other.targets) == 0:
            # This's because numpy does not treat array([], dtype=float64)
            # and array=array([], shape=(0, 4), dtype=float64) equally.
            return True
        return np.array_equal(self.array, other.array) and self.targets == other.targets
    def __getitem__(self, item) -> "Floats4InternalType":
        _arr = self.array[item].astype(float)
        if isinstance(item, int):
            _arr = _arr[np.newaxis, ...]
            _target = [self.targets[item]] if self.targets else []
        elif isinstance(item, slice):
            _target = self.targets[item]
        else:
            _target = [self.targets[i] for i in item] if self.targets else []
        return self.__class__(array=_arr, targets=_target)
```

Reply:
@Dipet 
By the way, why its called `Floats4InternalType`? If it represents a batch data with targets, then we should call it `TargetBatchData` or something like this?